mtags/src/main/scala-2/scala/meta/internal/pc/SignatureHelpProvider.scala

```diff
@@ -509,14 +527,28 @@ class SignatureHelpProvider(val compiler: MetalsGlobal)(implicit
             }
             paramss
           }
-        toSignatureInformation(
+
+        val methodType = if (!t.call.isUnapplyMethod) {
+          // It is not unapply.
+          tpe
+        } else if (isCaseClassConstructor) {
+          // It is unapply and case class.
+          tpe.resultType.typeSymbol.primaryConstructor.info
```

Comment:
This might be also loosing type information (though it might not be the cause of losing type information for param types)